### PR TITLE
Patch: Exclusion tables

### PR DESCRIPTION
**Pull request by joeflack4**
Updates:
```
    Feature: Exclusion tables
    - uri_to_curie: Refactored to a combo of my code plus OAK's after all 3 failed : my code by itself, oaklib, bioregistry.
```

### CODE REVIEW

**Comment by joeflack4** [comment]
This is a necessary patch. In my latest update before we merged, I didn't check all the results. ICD10WHO results, at least, should be incorrect until applying this.

File: src/scripts/exclusion_term_expansion.py
Comment (joeflack4) [inline src/scripts/exclusion_term_expansion.py:59]: I noticed this problem with angle brackets when I tried to use it the `bioregistry` feature you mentioned, so I opened up [an issue](https://github.com/biopragmatics/bioregistry/issues/479).

I know that eventually we want to use `oaklib`.

Comment (joeflack4) [reply src/scripts/exclusion_term_expansion.py:59]: *sigh*. Things take so much longer than expected. This `bioregistry` feature is too slow. I created an issue: https://github.com/biopragmatics/bioregistry/issues/480

I looked and the `oaklib` implementation only comes as class methods which require me to do a more elaborate setup and load up an ontology, and I don't have time to do that tonight if I want to produce some results. For now I'll borrow the `oak` code and repurpose it for both my PRs, but I'll need to change this soon.

File: src/scripts/exclusion_term_expansion.py
Comment (matentzn) [inline src/scripts/exclusion_term_expansion.py:69]: I would prefer not to have a default for when no prefix_map is provided. just let it fail hard in this case, rows 70-76 are very error prone guesswork and you can imagine dozens of cases where it is simply wrong

File: src/scripts/exclusion_term_expansion.py
Comment (matentzn) [inline src/scripts/exclusion_term_expansion.py:66]: why is this in the `if prefix_map` section of the code? It should be in the `else` part! If there is a prefix_map, that should always be preferred. Indeed, see comment below, I would prefer if we always use prefix_map, and raise an exception of the URI cannot be checked with the prefix_map. (prefix_map should be required).

**Review by matentzn — CHANGES_REQUESTED** [review]

File: src/scripts/exclusion_term_expansion.py
Comment (matentzn) [reply src/scripts/exclusion_term_expansion.py]: @joeflack4 no worries, it takes always longer to try and re-use code then to build it yourself for such methods. In the long term it will be worth, it, thank you for your patience! 

The ankle brackets issue: This was actually a mistake in _my_ code: I should have not served you IRIs with `<>`. It is good to know that these ankle brackets denotes "entity reference" is RDF. So, `http://purl.obolibrary.org/obo/MONDO_123` is an IRI, `<http://purl.obolibrary.org/obo/MONDO_123>` is a an entity reference in RDF (for example, a turtle or SPARQL file). I should removed these from the files i provided to you.

Seems the caching issue is also solved - for future iterations of the function please fold it in. :) Thanks!

File: src/scripts/exclusion_term_expansion.py
Comment (joeflack4) [reply src/scripts/exclusion_term_expansion.py:69]: That suggestion makes sense. I think there might have been some cases in the past where I didn't have a `prefix_map`, but that's not a situation we want to be in anyway, and I think I've gotten past that point too, so I can remove it.

File: src/scripts/exclusion_term_expansion.py
Comment (joeflack4) [reply src/scripts/exclusion_term_expansion.py:66]: I saw your comment [on Slack](https://monarchinitiative.slack.com/archives/C0222P0736F/p1659356271536479) about this as well. This was towards the end of a really looong day of work. I was just trying to get one of the libraries to work and clock out for the night. This code came from OAK, actually: https://github.com/INCATools/ontology-access-kit/blob/main/src/oaklib/interfaces/basic_ontology_interface.py#L136

Although there is a `TODO` in there to change that.

It's NP; I understand your logic completely. I'll remove it.

File: src/scripts/exclusion_term_expansion.py
Comment (joeflack4) [reply src/scripts/exclusion_term_expansion.py]: done!

File: src/scripts/exclusion_term_expansion.py
Comment (joeflack4) [reply src/scripts/exclusion_term_expansion.py]: done!

File: src/scripts/exclusion_term_expansion.py
Comment (joeflack4) [reply src/scripts/exclusion_term_expansion.py]: @matentzn Thanks, I understand. I did change the code to be even more robust though, if you want to leave this in:

Rather than a simple find/replace:
```py
    if uri.startswith('<'):
        uri = uri[1:]
    if uri.endswith('<'):
        uri = uri[:-1]
```

I can't think of any situation in which this would negatively effect things, unless we want to fail early and often, which I guess is maybe Pythonic.

I haven't utilized the `bioregistry` since it was updated a few days ago, but there's a `todo` in there for me to check out Charlie's new library. I don't think we should let that hold up a patch that will fix a bugfix though.

Comment (matentzn) [reply src/scripts/exclusion_term_expansion.py]: Ok to all. Thank you

File: src/scripts/exclusion_term_expansion.py
Comment (matentzn) [inline src/scripts/exclusion_term_expansion.py:60]:
```suggestion
    if uri.endswith('>'):
```

Comment (joeflack4) [reply src/scripts/exclusion_term_expansion.py:60]: Haha! Ooops. @_@

Comment (joeflack4) [reply src/scripts/exclusion_term_expansion.py:60]: Fixed

**Review by matentzn — APPROVED** [review]

**Pull request merged**
